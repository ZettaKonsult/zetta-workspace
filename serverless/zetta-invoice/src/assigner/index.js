/* @flow */

/**
 * @date 2018-01-16
 */
import getAssignments from './getAssignments';

export { getAssignments };
