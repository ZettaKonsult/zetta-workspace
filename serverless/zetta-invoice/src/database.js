/* @flow */

import AWS from 'aws-sdk';

const isOffline = process.env.IS_OFFLINE;
console.log(isOffline);

const dynamodbOfflineOptions = {
  region: 'localhost',
  endpoint: 'http://localhost:8000',
};

const client = isOffline
  ? new AWS.DynamoDB.DocumentClient(dynamodbOfflineOptions)
  : new AWS.DynamoDB.DocumentClient();

export const getDbTable = (params: { name: string }): string => {
  return params.name;
};

export default async (method: string, params: { [string]: any }) => {
  return await client[method](params).promise();
};
