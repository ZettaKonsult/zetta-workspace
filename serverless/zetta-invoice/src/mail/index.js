/* @flow */

/**
 * @date 2018-03-12
 */

import invoicePDF from './mail';
import { send, prepareAndSend, sendInvoice } from './emailDoc';
export default { invoicePDF, send, sendInvoice, prepareAndSend };
