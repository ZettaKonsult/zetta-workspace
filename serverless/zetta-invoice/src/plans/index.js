/* @flow */

/**
 * @date 2018-03
 */

import get from './get';
import list from './list';
import save from './save';
import remove from './remove';
import { updateRecipientIds } from './update';

export default { get, list, save, updateRecipientIds, remove };
