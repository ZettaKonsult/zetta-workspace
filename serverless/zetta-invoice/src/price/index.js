/**
 * @date 2018-04-10
 */

import calculate from './price';

export default { calculate };
