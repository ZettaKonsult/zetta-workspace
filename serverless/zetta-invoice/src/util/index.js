import parser from './parser';
import { getS3Object, putS3Object } from './s3';
import db, { getDbTable } from './database';
import { failure, success } from './response';
import request from './http';

export {
  parser,
  request,
  getS3Object,
  putS3Object,
  db,
  getDbTable,
  failure,
  success,
};
