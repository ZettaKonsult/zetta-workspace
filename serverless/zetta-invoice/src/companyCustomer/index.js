/* @flow */

import companyCustomerDatabase from './companyCustomerDatabase';
import { db, getDbTable } from '../util';

const companyCustomerTable = getDbTable({ name: 'CompanyCustomers' });

export default companyCustomerDatabase(db)(companyCustomerTable);
