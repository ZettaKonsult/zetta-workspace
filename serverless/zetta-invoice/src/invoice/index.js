/* @flow */

/**
 * @date 2017-02-13
 */
import invoice from './invoice';
import invoiceDatabase from './invoiceDynamodb';
import invoiceGroupDatabase from './invoiceGroupDatabase';

import { db } from '../util';
import { getDbTable } from '../util/database';

const TableNameInvoice = getDbTable({ name: 'Invoices' });
const databaseInvoice = invoiceDatabase(db)(TableNameInvoice);

const TableNameInvoiceGroup = getDbTable({ name: 'InvoiceGroups' });
const databaseInvoiceGroup = invoiceGroupDatabase(db)(TableNameInvoiceGroup);

export default {
  create: data => invoice().create(data),
  invoiceDatabase: databaseInvoice,
  databaseInvoiceGroup,
};
