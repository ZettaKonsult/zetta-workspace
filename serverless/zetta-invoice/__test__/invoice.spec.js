/**
 * @date 2018-03-05
 */
import cuid from 'cuid';
import { request, testConfig } from '../src/util/';

const host = testConfig.Host;

let companyCustomerId1 = cuid();
let companyCustomerId2 = cuid();
let invoices = [
  {
    companyCustomerId: companyCustomerId1,
    invoiceRows: [
      { description: '3an A', unit: 34, price: 345, tax: 0.25 },
      { description: '3an B', unit: 34, price: 345, tax: 0.25 },
      { description: '3an C', unit: 34, price: 345, tax: 0.25 },
    ],
    recipientIds: ['recipientId1'],
  },
  {
    companyCustomerId: companyCustomerId2,
    invoiceRows: [
      { description: '3an A', unit: 34, price: 345, tax: 0.25 },
      { description: '3an B', unit: 34, price: 345, tax: 0.25 },
      { description: '3an C', unit: 34, price: 345, tax: 0.25 },
    ],
    recipientIds: ['recipientId1'],
  },
];

beforeAll(async () => {
  let invoicesPromise = invoices.map(invoice =>
    request({
      host,
      path: `invoice`,
      payload: {
        method: 'post',
        body: invoice,
      },
    })
  );
  invoices = await Promise.all(invoicesPromise);
});

afterAll(async () => {
  let invoicesPromise = invoices.map(invoice =>
    request({
      host,
      path: 'invoice',
      payload: {
        method: 'delete',
        body: {
          companyCustomerId: invoice.companyCustomerId,
          invoiceId: invoice.id,
        },
      },
    })
  );
  await Promise.all(invoicesPromise);
});

describe('Invoices', () => {
  it('invoices have correct attributes after creation', () => {
    let invoice = invoices[0];

    expect(invoice).toHaveProperty('id');
  });

  it('get unlocked invoices', async () => {
    const result = await request({
      payload: {
        method: 'get',
      },
      host,
      path: `invoice/${companyCustomerId1}/false`,
    });

    expect(result).toEqual([invoices[0]]);
  });

  //TODO this endpoint does not inject companyCustomer on locking yet
  it.skip('lock invoice', async () => {
    const result = await request({
      payload: {
        method: 'post',
        body: {
          companyCustomerId: companyCustomerId2,
          invoiceId: invoices[1].id,
          lock: 'true',
        },
      },
      host,
      path: `invoice/lock`,
    });

    const lockedInvoices = await request({
      payload: {
        method: 'get',
      },
      host,
      path: `invoice/${companyCustomerId2}/true`,
    });

    expect(result).toEqual({ [invoices[1].id]: { locked: true } });
    expect(lockedInvoices[0]).toHaveProperty('companyCustomer');
  });
});

describe('Statuses.', () => {
  it('Get one.', async () => {
    const expected = {
      id: 'itemStatusId1',
      invoiceId: 'invoiceId1',
      createdAt: 3456789012,
      itemStatus: 'pending',
    };
    const result = await request({
      payload: {
        method: 'get',
      },
      host,
      path: 'invoice/status/companyCustomerId/invoiceId1',
    });
    expect(result).toEqual(expected);
  });
  it('Get many.', async () => {
    const expected = [
      {
        id: 'itemStatusId1',
        invoiceId: 'invoiceId1',
        createdAt: 3456789012,
        itemStatus: 'pending',
      },
      {
        id: 'itemStatusId2',
        invoiceId: 'invoiceId1',
        createdAt: 6789012345,
        itemStatus: 'succeeded',
      },
    ];
    const result = await request({
      payload: {
        method: 'get',
      },
      host,
      path: 'invoice/status/invoiceId1',
    });
    expect(result).toEqual(expected);
  });
});
