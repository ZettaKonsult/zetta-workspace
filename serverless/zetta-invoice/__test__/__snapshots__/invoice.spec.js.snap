// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoices create a recipient 1`] = `
Object {
  "address": "Road 234A",
  "city": "RecipientCity",
  "companyCustomerId": Any<String>,
  "createdAt": 2345678901,
  "email": "fiddep@telia.com",
  "firstName": "RecipientFirst",
  "id": Any<String>,
  "lastName": "RecipientLast",
  "mobile": "+46762345678",
  "ssn": "1234567890",
  "zipcode": "12345",
}
`;

exports[`Invoices create companyCustomer 1`] = `
Object {
  "VAT": "91050400356",
  "address": "testRoad 33",
  "bank": Object {
    "giro": "123-4567",
    "name": "MoneyBank",
  },
  "city": "testCity",
  "company": "zetta konsult",
  "email": "test@test.test",
  "firstName": "Test",
  "id": Any<String>,
  "lastName": "TestPersson",
  "mobile": "+46762345678",
  "ssn": "1234567890",
  "zipcode": "12345",
}
`;

exports[`Invoices invoices have correct attributes after creation 1`] = `
Object {
  "companyCustomerId": Any<String>,
  "createdAt": Any<Number>,
  "id": Any<String>,
  "invoiceRows": Array [
    Object {
      "description": "3an A",
      "price": 345,
      "tax": 0.25,
      "unit": 34,
    },
    Object {
      "description": "3an B",
      "price": 345,
      "tax": 0.25,
      "unit": 34,
    },
    Object {
      "description": "3an C",
      "price": 345,
      "tax": 0.25,
      "unit": 34,
    },
  ],
  "locked": false,
  "recipientIds": Array [
    "cjj8uxu5s005hw4cvs3xidsm7",
  ],
  "recipients": Array [
    "cjj8uxu5s005hw4cvs3xidsm7",
  ],
  "status": Array [],
}
`;
